src/LEGO.AsyncAPI/Models/Bindings/Sns/Statement.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using LEGO.AsyncAPI.Models.Interfaces;
+using LEGO.AsyncAPI.Writers;
+
+namespace LEGO.AsyncAPI.Models.Bindings.Sns;
+
+public class Statement : IAsyncApiElement
+{
+    public Effect Effect { get; set; }
+
+    /// <summary>
+    /// The AWS account or resource ARN that this statement applies to.
+    /// </summary>
+    // public StringOrStringList Principal { get; set; }
+    public string Principal { get; set; }
+
+    /// <summary>
+    /// The SNS permission being allowed or denied e.g. sns:Publish
+    /// </summary>
+    // public StringOrStringList Action { get; set; }
+
+    public void Serialize(IAsyncApiWriter writer)
+    {
+        if (writer is null)
+        {
+            throw new ArgumentNullException(nameof(writer));
+        }
+
+        writer.WriteStartObject();
+        // writer.WriteOptionalObject(AsyncApiConstants.Principal, this.Principal, (w, t) => t.Serialize(w));
+        writer.WriteOptionalProperty(AsyncApiConstants.Principal, this.Principal);
+        writer.WriteEndObject();
+    }
+}
+
+public enum Effect
+{
+    Allow,
+    Deny
+}
+
+public class StringOrStringList : IAsyncApiElement
+{
+    public string StringValue { get; set; }
+
+    public List<string> StringList { get; set; }
+}
```

Comment:
@VisualBean Question: these properties in the spec are JSON schema properties that can be a `string` or a `List<string>`. But I don't see a way using the current writers to parse these "dual" type properties. Is there a writer that can do that currently?

Reply:
What I've done previously is simply implement a "If length ==1 write string, else write array" You will see a similar thing in AsyncApiSchema

```sharp
            if (this.Type != null)
            {
                if (this.Type.Count == 1)
                {
                    writer.WriteOptionalProperty(AsyncApiConstants.Type, this.Type.First().GetDisplayName());
                }
                else
                {
                    writer.WriteOptionalCollection(AsyncApiConstants.Type, this.Type.Select(t => t.GetDisplayName()), (w, s) => w.WriteValue(s));
                }
            }
```